### PR TITLE
Add option to use ComboBoxes instead of Tabs and enable re-ordering of documents in the Documents Docklet

### PR DESCRIPTION
**Pull request by LucaDanieli**
Scott Wilson, Scott Carver and I developed the management of the documents in SuperCollider, connecting the Documents Docklet to the TabBar.

Here, 2 new options are added to Preferences->Editor :

- Use ComboBox
- Convert automatically to ComboBox when splitting

The documents docklet is dynamic and permits the re-ordering of documents. Furthermore, the TabBar and the Documents Docklet always show the documents in the same order, and any re-ordering in the TabBar affects the order of the documents in the docklet (and viceversa).

ComboBoxes are implemented to make clearer what is the document visualised by each split. Following a discussion in 2014, we have sorted each ComboBox alphabetically - to quickly find specific documents in performance.

In this way there are now 3 different types of ordering:

1) The TabBar and the Documents Docklet automatically order documents by "last opened", letting the user to reorder them;
2) Each editor has an optional ComboBox with documents sorted in Lexical Order;
3) ctrl+Tab / ctrl+shift+tab present a list of documents ordered by "last document used".

### CODE REVIEW

File: editors/sc-ide/widgets/doc_list.hpp [hunk 1]
```diff
@@ -91,6 +97,7 @@ class DocumentsDocklet : public Docklet
 private:
 
     DocumentListWidget *mDocList;
+
```
Comment (muellmusik) [inline editors/sc-ide/widgets/doc_list.hpp]: I'd lose this line, as it's just additional unnecessary whitespace

**Comment by telephon** [comment]
maybe solves this (?): #2612

**Comment by muellmusik** [comment]
@LucaDanieli, that's good. Before merging we should just do a clean PR which combines those adjustments with the original commits (I can show you how) in order to avoid polluting the commit log with no-ops.

Any other comments on this?

**Comment by snappizz** [comment]
I understand this mostly, but I'm not clear on what is meant by "Convert automatically to ComboBox when splitting." What is a "split"?

**Comment by LucaDanieli** [comment]
Hi Nathan.

Splitting is the ability to "split" the editor docklet into many editors, so to show different documents at once. You can find if in "view->split to right, view->split to bottom"

**Comment by snappizz** [comment]
Ok, makes sense. Thanks for your work on this. I've filed a PR (#2639) to reword some of the checkboxes so they're easier to understand at a glance.

**Comment by brianlheim** [comment]
I think this was merged prematurely. There are a couple of issues:

- The preferences options are vague, especially "Use ComboBox"
- The proper name for this graphical element is a [drop-down or drop-down list](https://msdn.microsoft.com/en-us/library/windows/desktop/dn742404(v=vs.85).aspx). 
- If I have two editors (one split) with the tab bar active and apply the "Use ComboBox" option, nothing happens.
- If I have "Use ComboBox" checked and not "Convert automatically to ComboBox when splitting" when I activate a split, I have a tab bar instead of combo boxes now.
- Is `Ctrl` in `Ctrl+Tab` and `Ctrl+Shift+Tab` meant to be `Cmd` on macOS? `Ctrl+Tab` does nothing for me.

**Comment by brianlheim** [comment]
Not sure how you want to fix this re: separate issues or PRs so I thought I would just raise them here first.

**Comment by LucaDanieli** [comment]
Thank you Brian for your review.

About the first two points: I am not English mothertongue and for me naming things is one of the most difficult things. I apologise.
About point 3 and 4: I can fix that now. Should I push to this branch? Or should I push to PR(#2639), Nathan?
About point 5: Yes, it is Cmd+Tab and Cmd+Shift+Tab

**Comment by snappizz** [comment]
@LucaDanieli I think you should just keep working on this branch.

**Comment by brianlheim** [comment]
@LucaDanieli 

> About the first two points: I am not English mothertongue and for me naming things is one of the most difficult things. I apologise.

No need to apologize, this is a fantastic feature and I'm really glad you guys added it! Text strings are the easiest things to change so it's no big deal. :)

> About point 5: Yes, it is Cmd+Tab and Cmd+Shift+Tab

Unfortunately these are macOS-specific commands for moving between applications, similar to `Alt+Tab` on Linux and Windows, so they won't be useful to most macOS users. In fact, you could probably just add macOS-specific code to change it to `Alt+(Shift+)Tab`--I don't think there are any common macOS commands with that shortcut.

**Comment by LucaDanieli** [comment]
@brianlheim 

The Alt+Tab was a shortcut already implemented, so I thought it was already working on different platforms. @muellmusik can you check what is the shortcut in Mac for switching among "last documents used"?

**Comment by LucaDanieli** [comment]
The fix is in this PR: #2641

**Comment by brianlheim** [comment]
@LucaDanieli @muellmusik My apologies, the shortcut for macOS is already [`Alt+(Shift+)Tab`](https://github.com/supercollider/supercollider/blob/master/editors/sc-ide/widgets/multi_editor.cpp#L639-L646), and I can see that now looking at the View menu in my running build. So that should be documented in the changelog, but there's nothing to change in the code. Thanks for helping to clear that up!
